File: MTGSearch/src/main/java/com/dbottillo/mtgsearchfree/model/helper/AddFavouritesAsyncTask.java
package com.dbottillo.mtgsearchfree.model.helper;

import android.content.Context;
import android.os.AsyncTask;
import android.widget.Toast;

import com.dbottillo.mtgsearchfree.model.database.CardDataSource;
import com.dbottillo.mtgsearchfree.model.database.CardsInfoDbHelper;
import com.dbottillo.mtgsearchfree.model.database.FavouritesDataSource;
import com.dbottillo.mtgsearchfree.model.database.MTGCardDataSource;
import com.dbottillo.mtgsearchfree.model.database.MTGDatabaseHelper;
import com.dbottillo.mtgsearchfree.model.MTGCard;
import com.google.gson.Gson;

import java.util.ArrayList;
import java.util.List;

/*
    this class is used only on debug to generate random favourites cards
 */
public class AddFavouritesAsyncTask extends AsyncTask<String, Void, ArrayList<Object>> {

    private boolean error = false;
    private Context context;

    public AddFavouritesAsyncTask(Context context) {
        this.context = context;
    }

    @Override
    protected ArrayList<Object> doInBackground(String... params) {
        ArrayList<Object> result = new ArrayList<>();

        MTGDatabaseHelper databaseHelper = new MTGDatabaseHelper(context);
        CardsInfoDbHelper cardsInfoDbHelper = new CardsInfoDbHelper(context);
        CardDataSource cardDataSource = new CardDataSource(cardsInfoDbHelper.getWritableDatabase(), new Gson());
        MTGCardDataSource mtgCardDataSource = new MTGCardDataSource(databaseHelper.getReadableDatabase(), cardDataSource);

        FavouritesDataSource favouritesDataSource = new FavouritesDataSource(cardsInfoDbHelper.getWritableDatabase(), cardDataSource);
        favouritesDataSource.clear();
        List<MTGCard> cards = mtgCardDataSource.getRandomCard(600);
        for (MTGCard card : cards) {
            favouritesDataSource.saveFavourites(card);
        }

        return result;
    }

    @Override
    protected void onPostExecute(ArrayList<Object> result) {
        if (error) {
            Toast.makeText(context, "error", Toast.LENGTH_SHORT).show();
        } else {
            Toast.makeText(context, "finished", Toast.LENGTH_SHORT).show();
        }
    }

}

File: MTGSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/model/helper/AddFavouritesAsyncTask.kt
package com.dbottillo.mtgsearchfree.model.helper

import android.content.Context
import android.os.AsyncTask
import android.widget.Toast
import com.dbottillo.mtgsearchfree.model.storage.database.CardDataSource
import com.dbottillo.mtgsearchfree.model.database.CardsInfoDbHelper
import com.dbottillo.mtgsearchfree.model.storage.database.FavouritesDataSource
import com.dbottillo.mtgsearchfree.model.database.MTGCardDataSource
import com.dbottillo.mtgsearchfree.model.database.MTGDatabaseHelper
import com.google.gson.Gson
import java.lang.ref.WeakReference

/*
    This class is used only on debug to generate random favourites cards
 */
class AddFavouritesAsyncTask(context: Context) : AsyncTask<String, Void, ArrayList<Any>>() {

    private val context: WeakReference<Context> = WeakReference(context)

    override fun doInBackground(vararg params: String): ArrayList<Any> {
        val result = ArrayList<Any>()

        context.get()?.let {
            val databaseHelper = MTGDatabaseHelper(it)
            val cardsInfoDbHelper = CardsInfoDbHelper(it)
            val cardDataSource = CardDataSource(cardsInfoDbHelper.writableDatabase, Gson())
            val mtgCardDataSource = MTGCardDataSource(databaseHelper.readableDatabase, cardDataSource)

            val favouritesDataSource = FavouritesDataSource(cardsInfoDbHelper.writableDatabase, cardDataSource)
            favouritesDataSource.clear()

            mtgCardDataSource.getRandomCard(600).forEach {
                favouritesDataSource.saveFavourites(it)
            }

            cardsInfoDbHelper.close()
            databaseHelper.close()
            return result
        } ?: return arrayListOf()
    }

    override fun onPostExecute(result: ArrayList<Any>) {
        context.get()?.let {
            Toast.makeText(it, "finished", Toast.LENGTH_SHORT).show()
        }
    }
}
